sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClient.java

```diff
@@ -110,7 +222,14 @@ public TableServiceVersion getServiceVersion() {
      * {@link AzureNamedKeyCredential}.
      */
     public String generateSas(TableSasSignatureValues tableSasSignatureValues) {
-        return client.generateSas(tableSasSignatureValues);
+        AzureNamedKeyCredential azureNamedKeyCredential = TableSasUtils.extractNamedKeyCredential(getHttpPipeline());
+
+        if (azureNamedKeyCredential == null) {
+            throw logger.logExceptionAsError(new IllegalStateException("Cannot generate a SAS token with a client that"
```

Comment:
Consider putting these strings into util.Constants so they can be shared with asyn client.